shared/sys/sys_win32.cpp

```diff
@@ -590,3 +591,87 @@ void Sys_Sleep( int msec )
 	Sleep( msec );
 #endif
 }
+
+/*
+================
+Sys_SteamInit
+
+Steam initialization is done here.
+In order for Steam to work, two things are needed:
+- steam_api.dll (not included with retail Jedi Academy or Jedi Outcast!)
+- steam_appid.txt (likewise)
+steam_appid.txt is a text file containing either "6020" or "6030".
+These correspond to Jedi Academy and Jedi Outcast, respectively.
+
+Steamworks SDK is required to use the playtime tracking and overlay features
+without launching the app manually through Steam.
+Unfortunately, the SDK does not play nice with copyleft licenses.
+Fortunately! we can invoke the library directly and avoid this entirely,
+provided the end-user has the goods.
+Unfortunately! this is platform specific and so we have to do it here.
+================
+*/
+
+typedef bool(__stdcall* SteamAPIInit_Type)();
+typedef void(__stdcall* SteamAPIShutdown_Type)();
+static SteamAPIInit_Type SteamAPI_Init;
+static SteamAPIShutdown_Type SteamAPI_Shutdown;
+static void* gp_steamLibrary = nullptr;
+
+void Sys_SteamInit()
+{
+	if (!Cvar_VariableIntegerValue("com_steamIntegration"))
+	{
+		// Don't do anything if com_steamIntegration is disabled
+		return;
+	}
+
+	// Load the library
+	gp_steamLibrary = Sys_LoadLibrary("steam_api" DLL_EXT);
+	if (!gp_steamLibrary)
+	{
+		Com_Printf(S_COLOR_RED "Steam integration failed: Couldn't find steam_api" DLL_EXT "\n");
+		return;
+	}
+
+	// Load the functions
+	SteamAPI_Init = (SteamAPIInit_Type)Sys_LoadFunction(gp_steamLibrary, "SteamAPI_Init");
+	SteamAPI_Shutdown = (SteamAPIShutdown_Type)Sys_LoadFunction(gp_steamLibrary, "SteamAPI_Shutdown");
+
+	if (!SteamAPI_Shutdown || !SteamAPI_Init)
+	{
+		Com_Printf(S_COLOR_RED "Steam integration failed: Library invalid\n");
+		Sys_UnloadLibrary(gp_steamLibrary);
+		gp_steamLibrary = nullptr;
+		return;
+	}
+
+	// Finally, call the init function in Steam, which should pop up the overlay if everything went correctly
+	if (!SteamAPI_Init())
+	{
+		Com_Printf(S_COLOR_RED "Steam integration failed: Steam init failed. Ensure steam_appid.txt exists and is valid.\n");
+		Sys_UnloadLibrary(gp_steamLibrary);
+		gp_steamLibrary = nullptr;
+		return;
+	}
+}
+
+/*
+================
+Sys_SteamShutdown
+
+Platform-specific exit code
+================
+*/
+void Sys_SteamShutdown()
+{
+	if (!gp_steamLibrary)
+	{
+		Com_Printf("Skipping Steam integration shutdown...\n");
+		return;
+	}
+
+	SteamAPI_Shutdown();
+	Sys_UnloadLibrary(gp_steamLibrary);
+	gp_steamLibrary = nullptr;
+}
```

Comment:
Missing new line at EOF. Not going to cause any problems but it's best practices to include this